source/reflect.h2

```diff
@@ -1371,6 +1372,66 @@ print: (t: meta::type_declaration) =
     std::cout << t.print() << "\n";
 }
 
+//-----------------------------------------------------------------------
+//
+//  regex - creates regular expressions from members
+//
+//  TODO
+//
+regex_gen: (inout t: meta::type_declaration) =
+{
+    has_default: bool = false;
+    expressions: std::map<std::string, std::string> = ();
+
+    for t.get_member_objects() do (inout m)
+    {
+        name: std::string = m.name();
+        prefix: std::string = "regex";
+
+        if name.starts_with(prefix) {
+
+            if !m.has_initializer() {
+                t.error("Regular expression must have an initializer.");
+            }
+            m.mark_for_removal_from_enclosing_type();
+
+            name = name.substr(prefix.size());
+
+            if name.starts_with("_") {
+                name = name.substr(1);
+            }
+
+            if name.empty() {
+                if has_default {
+                    t.error("Type can only contain one default named regular expression.");
+                }
+                has_default = true;
+            }
+
+            expr: std::string = m.initializer();
+            if !(expr.starts_with("\"") && expr.ends_with("")) {
```

Comment:
How can a string end with an empty string?  Should that be `"\""` as well?

Reply:
I find it easier to read `if !x || !y` vs `if !(x && y)`.  Others may have differing opinions, not sure if there is an existing style here.

Reply:
```suggestion
            if !expr.is_quoted() {
```

Reply:
Of course.

Reply:
Done.